askama_parser/src/expr.rs

```diff
@@ -300,6 +303,74 @@ impl<'a> Expr<'a> {
     fn char(i: &'a str) -> ParseResult<'a, Self> {
         map(char_lit, Self::CharLit)(i)
     }
+
+    #[cfg(not(feature = "i18n"))]
+    fn localize(i: &'a str) -> ParseResult<'a, Self> {
+        let (i, _) = pair(tag("localize"), ws(tag("(")))(i)?;
+        eprintln!(r#"Activate the "i18n" feature to use {{ localize() }}."#);
```

Comment:
This should not be an `eprintln!()`, but some form of error that is yielded to the caller. (Similar for the other error conditions here.)

Reply:
like so?

```diff
     #[cfg(not(feature = "i18n"))]
     fn localize(i: &'a str) -> ParseResult<'a, Self> {
+        // Happy path: no i18n _should_ fail to parse "localize ("
         let (i, _) = pair(tag("localize"), ws(tag("(")))(i)?;
-        eprintln!(r#"Activate the "i18n" feature to use {{ localize() }}."#);
-        Err(nom::Err::Failure(error_position!(i, ErrorKind::Tag)))
+
+        Err(nom::Err::Failure(ErrorContext {
+            input: i,
+            message: Some(Cow::Owned(format!(
+                r#"Activate the "i18n" feature to use {{ localize() }}."#
+            )))
+        }))
     }
```

Reply:
Yup, that looks reasonable to me!